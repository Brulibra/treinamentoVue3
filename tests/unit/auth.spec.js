import mockAxios from "axios";
import AuthService from "../../src/services/auth";

jest.mock("axios");

describe("AuthService", () => {
  it("should return a token when user login", async () => {
    const token = "123.123.123";
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: { token } });
    });

    const response = await AuthService(mockAxios).login({
      email: "teste@email.com",
      password: "123",
    });
    expect(response.data).toHaveProperty("token");
    expect(response).toMatchSnapshot()
  });
});
