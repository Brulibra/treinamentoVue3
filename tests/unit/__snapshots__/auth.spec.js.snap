// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthService should return a token when user log-in 1`] = `
{
  "data": {
    "token": "123.123.123",
  },
  "errors": null,
}
`;

exports[`AuthService should return an user when register 1`] = `
{
  "data": {
    "email": "teste@email.com",
    "name": "Ana",
    "password": "123",
  },
  "errors": null,
}
`;
